pyflagser/flagser.py

```diff
@@ -86,18 +92,30 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
     if not approximation:
         approximation = -1
 
-    edges = sp.coo_matrix(flag_matrix, copy=True)
-    edges.setdiag(np.nan)
+    if type(flag_matrix) is np.ndarray:
```

Comment:
A little surprised to not see `isinstance(flag_matrix, np.ndarray)` here which I thought would be the recommended syntax.